python/ray/function_manager.py

```diff
@@ -509,20 +509,27 @@ def _publish_actor_class_to_key(self, key, actor_class_info):
         """
         # We set the driver ID here because it may not have been available when
         # the actor class was defined.
-        actor_class_info["driver_id"] = self._worker.task_driver_id.id()
         self._worker.redis_client.hmset(key, actor_class_info)
         self._worker.redis_client.rpush("Exports", key)
 
     def export_actor_class(self, Class, actor_method_names,
                            checkpoint_interval):
         function_descriptor = FunctionDescriptor.from_class(Class)
-        key = (b"ActorClass:" + self._worker.task_driver_id.id() + b":" +
+        if (self._worker.task_driver_id.is_nil()):
+            logger.warning("export_actor_class with NIL task_driver_id, this "
+                           "may happen when export_actor_class runs not in "
+                           "the main thread. Will wait for the driver id.")
+        while (self._worker.task_driver_id.is_nil()):
+            time.sleep(0.01)
```

Comment:
Assume I have an actor that only creates background threads to do work. Its main thread never receives tasks. 
Sleeping here will block forever. 
Instead of sleeping here, I think an easier solution is not resetting `task_driver_id` for actors. Because:
1) If a worker is an actor, all the tasks should have the same driver id.
2) If I have a background thread whose lifecycle expands across multiple main-thread tasks, the worker must be an actor. (Normal tasks shouldn't create a thread whose lifecycle is longer than itself. We can raise an error in this case.)

Reply:
That sounds right. If this is happening on an actor, then we should be able to figure out the correct driver ID. If it's happening on a non-actor worker, then we should raise an error.